### PR TITLE
Update ODR to 2.0, improve Dread exporter tests

### PR DESCRIPTION
**Pull request by henriquegemignani**

### CODE REVIEW

File: test/games/dread/exporter/test_dread_patch_data_factory.py [hunk 1]
```diff
@@ -220,20 +222,19 @@ def setup_and_teardown_for_custom_spawn():
...
-    # only one spawn point should be created
-    expected_new_spawn = [{'new_actor': {'actor': 'SP_RDV_000', 'scenario': 's010_cave'},
-                            'location': {'x': 6492.78, 'y': -9500.0, 'z': 0.0},
-                            'collision_camera_name': 'collision_camera_001'}]
-    expected_start = {'scenario': 's010_cave', 'actor': 'SP_RDV_000'}
-    assert data["new_spawn_points"] == expected_new_spawn
-    assert data["starting_location"] == expected_start
+
+    # Expected Result
+    with test_files_dir.joinpath("patcher_data", "dread", "custom_start.json").open("r") as file:
```
Comment (ThanatosGit) [inline test/games/dread/exporter/test_dread_patch_data_factory.py:237]: Same as above

**Review by ThanatosGit — CHANGES_REQUESTED** [review]

File: test/games/dread/exporter/test_dread_game_exporter.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from unittest.mock import MagicMock, ANY
+
+import open_dread_rando.dread_patcher
+import pytest
+
+from randovania.games.dread.exporter.game_exporter import DreadGameExporter, DreadGameExportParams, DreadModPlatform
+from randovania.lib import json_lib
+
+
+@pytest.mark.parametrize("patch_data_name", ["starter_preset", "custom_start"])
```
[the remark this answers is not in the document]
Comment (henriquegemignani) [reply test/games/dread/exporter/test_dread_game_exporter.py:11]: I have so much junk in my repository that new files hide among all the stuff I don't commit 😓

File: test/games/dread/exporter/test_dread_game_exporter.py [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+    mock_patch: MagicMock = mocker.patch("open_dread_rando.patch_with_status_update", side_effect=validate_schema)
+
+    patch_data = json_lib.read_path(test_files_dir.joinpath("patcher_data", "dread", f"{patch_data_name}.json"))
+
+    exporter = DreadGameExporter()
+    export_params = DreadGameExportParams(
+        spoiler_output=None,
+        input_path=tmp_path.joinpath("input_path"),
+        output_path=tmp_path.joinpath("output/path"),
```
Comment (ThanatosGit) [inline test/games/dread/exporter/test_dread_game_exporter.py:26]: Is there a specific reason we use a `/` here? Could be `joinpath("output", "path")` in that case?

Comment (henriquegemignani) [reply test/games/dread/exporter/test_dread_game_exporter.py:26]: Forgot and went with the easy way. Tbf for all our uses of paths, both separators work fine on Windows so it won't break anything.

**Pull request merged**